sklearn/discriminant_analysis.py

```diff
@@ -623,11 +623,15 @@ class QuadraticDiscriminantAnalysis(BaseEstimator, ClassifierMixin):
         Discriminant Analysis
     """
 
-    def __init__(self, priors=None, reg_param=0., store_covariances=False,
-                 tol=1.0e-4):
+    def __init__(self, priors=None, reg_param=0.,store_covariance=False, tol=1.0e-4,store_covariances=None):
+        if store_covariances is not None:
+        	warnings.warn("'store_covariances' was renamed to 'store_covariance' in version 0.17 and "
+                      "will be removed in later versions.", DeprecationWarning)
+        	self.store_covariance = store_covariances
```

Comment:
I think you should handle the deprecation in `fit`, not here. In scikit-learn, `__init__` is just used to store parameters, otherwise, in the general case, we will break parameter searches.

In `fit` set a local copy of `store_covariance` on the basis of both old and new params.